src/elements/CFbend.H

```diff
@@ -119,67 +168,44 @@ namespace impactx::elements
-
-            // access reference particle values to find beta*gamma^2
-            amrex::ParticleReal const pt_ref = refpart.pt;
-            amrex::ParticleReal const betgam2 = std::pow(pt_ref, 2) - 1.0_prt;
-            amrex::ParticleReal const bet = std::sqrt(betgam2/(1.0_prt + betgam2));
-
             // update horizontal and longitudinal phase space variables
-            amrex::ParticleReal const gx = m_k + std::pow(m_rc,-2);
-            amrex::ParticleReal const omegax = std::sqrt(std::abs(gx));
-
-            if(gx > 0.0) {
-                // calculate expensive terms once
-                auto const [sinx, cosx] = amrex::Math::sincos(omegax * slice_ds);
-                amrex::ParticleReal const r56 = slice_ds/betgam2
-                    + (sinx - omegax*slice_ds)/(gx*omegax * std::pow(bet,2) * std::pow(m_rc,2));
+            if (m_gx > 0.0)
+            {
+                amrex::ParticleReal const r56 = m_slice_ds * m_ibetgam2
+                    + (m_sinx - m_omega_x * m_slice_ds) * m_igobr;
 
                 // advance position and momentum (focusing)
-                x = cosx*xout + sinx/omegax*px - (1.0_prt - cosx)/(gx*bet*m_rc)*pt;
-                pxout = -omegax*sinx*xout + cosx*px - sinx/(omegax*bet*m_rc)*pt;
+                x = m_cosx * xout + m_sinx / m_omega_x * px - (1.0_prt - m_cosx) * m_igbrc * pt;
+                pxout = -m_omega_x * m_sinx * xout + m_cosx * px - m_sinx * m_iobrc * pt;
 
-                y = sinx/(omegax*bet*m_rc)*xout + (1.0_prt - cosx)/(gx*bet*m_rc)*px
-                    + tout + r56*pt;
+                y = m_sinx * m_iobrc * xout + (1.0_prt - m_cosx) * m_igbrc * px
+                    + tout + r56 * pt;
                 ptout = pt;
-            } else {
-                // calculate expensive terms once
-                amrex::ParticleReal const sinhx = std::sinh(omegax * slice_ds);
-                amrex::ParticleReal const coshx = std::cosh(omegax * slice_ds);
-                amrex::ParticleReal const r56 = slice_ds/betgam2
-                    + (sinhx - omegax*slice_ds)/(gx*omegax * std::pow(bet,2) * std::pow(m_rc,2));
+            } else
+            {
+                amrex::ParticleReal const r56 = m_slice_ds * m_ibetgam2
+                    + (m_sinhx - m_omega_x * m_slice_ds) * m_igobr;
 
                 // advance position and momentum (defocusing)
-                x = coshx*xout + sinhx/omegax*px - (1.0_prt - coshx)/(gx*bet*m_rc)*pt;
-                pxout = omegax*sinhx*xout + coshx*px - sinhx/(omegax*bet*m_rc)*pt;
+                x = m_coshx * xout + m_sinhx / m_omega_x * px - (1.0_prt - m_coshx) * m_igbrc * pt;
+                pxout = m_omega_x * m_sinhx * xout + m_coshx * px - m_sinhx * m_iobrc * pt;
 
-                t = sinhx/(omegax*bet*m_rc)*xout + (1.0_prt - coshx)/(gx*bet*m_rc)*px
-                    + tout + r56*pt;
+                t = m_sinhx * m_iobrc * xout + (1.0_prt - m_coshx) * m_igbrc * px
+                    + tout + r56 * pt;
```

Comment:
Something is still broken for the long variable, but I cannot spot it right now.

Reply:
Ah, found it. Used `m_gx` before assignment.